scripts/tools/retry.sh

```diff
@@ -9,13 +9,49 @@ function retry {
     local delay="${DELAY:-5}"
     local i=0
 
+    local cr
+    local nl
+    local isa_tty="false"
+    if [ -t 1 ]; then
+        isa_tty="true"
+        # Output is to a TTY; don't scroll display while waiting.
+        cr="\r"
+        # Display a trailing space between the last character and the cursor.
+        nl=" "
+    else
+        # Output is to a file; don't overwrite lines.
+        cr=""
+        nl="\n"
+    fi
+
+    local output
+    output="$(mktemp)"
```

Comment:
Hmm, we _could_ be leaving the temporary file behind if we get `^C`ed at the correct time… but I'm not sure I care.